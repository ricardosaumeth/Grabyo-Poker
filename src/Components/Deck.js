import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Card, CardDeck } from '../Styles/Styled';

export class Deck extends Component {
  render() {
    const { cardsDeck, allocatedDeck } = this.props;

    let cards = [];
    if (Array.isArray(cardsDeck) && cardsDeck) {
      let chunkOfCards1 = [];
      let chunkOfCards2 = [];
      let chunkOfCards3 = [];
      let chunkOfCards4 = [];

      for (let i = 0; i < 13; i++) {
        let isSelected = false;
        if (Array.isArray(allocatedDeck) && allocatedDeck) {
          for (let j = 0; j < allocatedDeck.length; j++) {
            for (let k = 0; k < 5; k++) {
              if (cardsDeck[i].name === allocatedDeck[j][k].name) {
                isSelected = true;
              }
            }
          }
        }

        chunkOfCards1.push(
          <Card
            key={cardsDeck[i].suit + cardsDeck[i].value}
            suit={cardsDeck[i].suit}
            value={cardsDeck[i].value}
            selected={isSelected}
          >
            {cardsDeck[i].value}
          </Card>
        );
      }

      for (let i = 13; i < 26; i++) {
        let isSelected = false;
        if (Array.isArray(allocatedDeck) && allocatedDeck) {
          for (let j = 0; j < allocatedDeck.length; j++) {
            for (let k = 0; k < 5; k++) {
              if (cardsDeck[i].name === allocatedDeck[j][k].name) {
                isSelected = true;
              }
            }
          }
        }
        chunkOfCards2.push(
          <Card
            key={cardsDeck[i].suit + cardsDeck[i].value}
            suit={cardsDeck[i].suit}
            value={cardsDeck[i].value}
            selected={isSelected}
          >
            {cardsDeck[i].value}
          </Card>
        );
      }
      for (let i = 26; i < 39; i++) {
        let isSelected = false;
        if (Array.isArray(allocatedDeck) && allocatedDeck) {
          for (let j = 0; j < allocatedDeck.length; j++) {
            for (let k = 0; k < 5; k++) {
              if (cardsDeck[i].name === allocatedDeck[j][k].name) {
                isSelected = true;
              }
            }
          }
        }
        chunkOfCards3.push(
          <Card
            key={cardsDeck[i].suit + cardsDeck[i].value}
            suit={cardsDeck[i].suit}
            value={cardsDeck[i].value}
            selected={isSelected}
          >
            {cardsDeck[i].value}
          </Card>
        );
      }
      for (let i = 39; i < 52; i++) {
        let isSelected = false;
        if (Array.isArray(allocatedDeck) && allocatedDeck) {
          for (let j = 0; j < allocatedDeck.length; j++) {
            for (let k = 0; k < 5; k++) {
              if (cardsDeck[i].name === allocatedDeck[j][k].name) {
                isSelected = true;
              }
            }
          }
        }
        chunkOfCards4.push(
          <Card
            key={cardsDeck[i].suit + cardsDeck[i].value}
            suit={cardsDeck[i].suit}
            value={cardsDeck[i].value}
            selected={isSelected}
          >
            {cardsDeck[i].value}
          </Card>
        );
      }

      const Deck = (
        <CardDeck key="cardDeck">
          <div key="1">{chunkOfCards1}</div>
          <div key="2">{chunkOfCards2}</div>
          <div key="3">{chunkOfCards3}</div>
          <div key="4">{chunkOfCards4}</div>
        </CardDeck>
      );

      cards.push(Deck);
    }

    return cards;
  }
}

const mapStateToProps = state => {
  return {
    cardsDeck: state.deck.cardsDeck,
    allocatedDeck: state.deck.allocatedDeck
  };
};

export default connect(mapStateToProps)(Deck);
