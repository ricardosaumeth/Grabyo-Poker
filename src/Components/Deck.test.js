import { mount, shallow } from "enzyme";
import React from "react";
import "../setupTests"

import { Deck } from "./Deck";

import renderer from 'react-test-renderer';

import DeckGenerator from '../GameLogic/deck';

describe(`Card deck`, () => {
	test('should match the Snapshot', () => {
		DeckGenerator.generate_deck();
		const deckCards = DeckGenerator.deck;
		const wrapper = renderer.create(<Deck cardsDeck={deckCards} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	test('should call Deckclass and create the deck', () => {
		DeckGenerator.generate_deck();
		const deckCards = DeckGenerator.deck;
		expect(deckCards).toContainEqual({ name: '2 of D', suit: 'D', value: '2', isSelected: false });
	});
});
