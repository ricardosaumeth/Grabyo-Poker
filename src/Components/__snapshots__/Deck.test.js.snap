// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card deck should match the Snapshot 1`] = `
<div
  className="sc-bZQynM ddCsBS"
>
  <div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="2"
    >
      2
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="3"
    >
      3
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="4"
    >
      4
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="5"
    >
      5
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="6"
    >
      6
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="7"
    >
      7
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="8"
    >
      8
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="9"
    >
      9
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="T"
    >
      T
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="J"
    >
      J
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="Q"
    >
      Q
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="K"
    >
      K
    </div>
    <div
      className="sc-htpNat lmaxQV"
      selected={false}
      value="A"
    >
      A
    </div>
  </div>
  <div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="2"
    >
      2
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="3"
    >
      3
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="4"
    >
      4
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="5"
    >
      5
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="6"
    >
      6
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="7"
    >
      7
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="8"
    >
      8
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="9"
    >
      9
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="T"
    >
      T
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="J"
    >
      J
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="Q"
    >
      Q
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="K"
    >
      K
    </div>
    <div
      className="sc-htpNat jXZeqR"
      selected={false}
      value="A"
    >
      A
    </div>
  </div>
  <div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="2"
    >
      2
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="3"
    >
      3
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="4"
    >
      4
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="5"
    >
      5
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="6"
    >
      6
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="7"
    >
      7
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="8"
    >
      8
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="9"
    >
      9
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="T"
    >
      T
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="J"
    >
      J
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="Q"
    >
      Q
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="K"
    >
      K
    </div>
    <div
      className="sc-htpNat egytKp"
      selected={false}
      value="A"
    >
      A
    </div>
  </div>
  <div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="2"
    >
      2
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="3"
    >
      3
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="4"
    >
      4
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="5"
    >
      5
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="6"
    >
      6
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="7"
    >
      7
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="8"
    >
      8
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="9"
    >
      9
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="T"
    >
      T
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="J"
    >
      J
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="Q"
    >
      Q
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="K"
    >
      K
    </div>
    <div
      className="sc-htpNat bgQLPz"
      selected={false}
      value="A"
    >
      A
    </div>
  </div>
</div>
`;
